File: app/src/main/java/com/melean/taskstimetracker/ui/interfaces/AddDataContract.java
package com.melean.taskstimetracker.ui.interfaces;

import com.melean.taskstimetracker.data.models.EmployeeModel;
import com.melean.taskstimetracker.data.models.TaskModel;

public interface AddDataContract {
    interface View {
        void showAddEmployeeView();

        void showAddTaskView();

        void showErrorAdding();

        void updateEmployees(EmployeeModel employee);

        void updateTasks(TaskModel task);

        void onDataEntered(Object obj);
    }

    interface UserActionsListener {
        void addEmployee();

        void addTaskView();

        void saveEmployee(EmployeeModel employee);

        void saveTask(TaskModel task);
    }
}

File: app/src/main/java/com/melean/taskstimetracker/ui/interfaces/AddDataContract.kt
package com.melean.taskstimetracker.ui.interfaces

import com.melean.taskstimetracker.data.models.EmployeeModel
import com.melean.taskstimetracker.data.models.TaskModel

interface AddDataContract {
    interface View {
        fun showAddEmployeeView()

        fun showAddTaskView()

        fun showErrorAdding()

        fun updateEmployees(employee: EmployeeModel)

        fun updateTasks(task: TaskModel)
    }

    interface UserActionsListener {
        fun addEmployee()

        fun addTaskView()

        fun saveEmployee(employee: EmployeeModel)

        fun saveTask(task: TaskModel)
    }
}